File: Leetcode-Problems/Rearrange Characters to Make Target String.cpp
class Solution
{
public:
    int rearrangeCharacters(string s, string target)
    {
        int ans = INT_MAX;
        unordered_map<char, int> mp1;
        unordered_map<char, int> mp2;
        for (char i : s)
        {
            mp1[i]++;
        }
        for (char i : target)
        {
            mp2[i]++;
        }
        for (auto i : mp2)
        {
            ans = min(ans, mp1[i.first] / i.second);
        }
        return ans;
    }
};

File: Leetcode-Problems/Rearrange Characters to Make Target String.java
class Solution {
    public int rearrangeCharacters(String s, String target) {
        int arrS[] = new int[26];
        int arrT[] = new int[26];
        for (char c : s.toCharArray()) {
            arrS[c - 'a']++;
        }
        for (char c : target.toCharArray()) {
            arrT[c - 'a']++;
        }
        int ans = Integer.MAX_VALUE;
        for (int i=0;i<26;i++) {
            if (arrT[i] != 0)
                ans = Math.min(ans, arrS[i] / arrT[i]);
        }
        return ans;
    }
}